addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2417,12 +2418,49 @@ const SnippetOptionWidget = Widget.extend({
                 await this[methodName](previewMode, widgetValue, params);
             }
         }
-
+        this._updateLockActive ^= 1;
         // We trigger the event on elements targeted by apply-to if any as
         // this.$target could not be in an editable element while the elements
         // targeted by apply-to are.
         ($applyTo || this.$target).trigger('content_changed');
     },
+    /**
+     * Activates the option associated to the given DOM element.
+     *
+     * @private
+     * @param {boolean|string} previewMode
+     *        - truthy if the option is enabled for preview or if leaving it (in
+     *          that second case, the value is 'reset')
+     *        - false if the option should be activated for good
+     * @param {UserValueWidget} widget - the widget which triggered the option change
+     * @returns {Promise}
+     */
+    _select: async function (previewMode, widget) {
+        const optionMethodsPromise = this._callOptionMethods(previewMode, widget);
+
+        if (!previewMode) {
+            const timeout = new Promise((resolve, reject) => {
+                const timer = setTimeout(() => {
+                    clearTimeout(timer);
+                    if (!!this._updateLockActive) {
+                        this.trigger_up('start_loading_overlay', {
+                            message: $('<i/>', {
+                                class: 'fa fa-spinner fa-pulse fa-3x text-white',
+                            }),
+                            css: {
+                                border: 'none',
+                                backgroundColor: 'transparent',
+                            }
+                        });
+                }}, 1000);
+            });
+            // will always resolve to 'optionMethodsPromise'
+            await Promise.race([optionMethodsPromise, timeout]);
+            this.trigger_up('end_loading_overlay');
+        } else {
+            await optionMethodsPromise;
+        }
+    },
```

Comment:
You don't need to change anything at the options level. Everything can be done in the `SnippetsMenu` class. When any change from the panel is being done, this goes through the event `'snippet_edition_request'` to ask the `SnippetsMenu` class to mutex-ualize the operation, so everything can be done there. This actually should be even better functionnaly by doing it there: if 3 options are done at around the same time and that each takes 800ms to be done (so the total is 2.4s) you have to see the loading during 1.4s. Doing this at option level like you did will not make it appear in that case.

In fact any operation which can take time is mutex-ualized by the `SnippetsMenu`. Any such operation should be subject to the loading screen: move each call to `_mutex.exec` into a function which will be in charge of doing the call to exec + adding/removing the loading screen when necessary